### PR TITLE
Feature/particle filter

### PR DESCRIPTION
**Pull request by yappel**
Added a filter to make an educated guess about the Pose of the user given the sensor data. It uses the "particle" algorithm to make this guess. A lot of finetuning of the parameters needs to be done to improve the performance of the filter.

The test TestParticleFilterRun might fail sometimes because of the randomness of the algorithm and the parameters are not optimized.

### CODE REVIEW

**Comment by lkrombeen** [comment]
Looks good, but you are missing some tests and 2 failed, which I fixed,

image

Also test the empty constructor of Pose: Pose() and the empty constructor of RotationMatrix: RotationMatrix()

**Comment by lkrombeen** [comment]
any reason for including the .xml's?

**Comment by lkrombeen** [comment]
Looks alrightyright

File: IRescue/UserLocalisation/Particle/Algos/Resamplers/MultinomialResampler.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        /// <summary>
+        /// Selects a list of indexes of the Particles that will survive the resampling.
+        /// </summary>
+        /// <param name="weights">The Weights of the Particles in a certain dimension</param>
+        /// <returns>The list with indexes of the Particles that are chosen by the resample algorithm</returns>
+        private static int[] Multinomial(Vector<float> weights)
+        {
+            int[] listout = new int[weights.Count];
+            CumSum(weights);
+            Random random = new Random();
```
Comment (remcodevos) [inline IRescue/UserLocalisation/Particle/Algos/Resamplers/MultinomialResampler.cs]: Why create a new Random each time that this function is called? Isn't this unnecessary and extra object creation time. Secondly, Random objects (according to documentation) created within 15ms of each other output the same values due to the way the seed is provided, this would remove part of the randomness.
